mypy/checker.py

```diff
@@ -4739,9 +4762,21 @@ def visit_decorator(self, e: Decorator) -> None:
             self.check_incompatible_property_override(e)
         # For overloaded functions we already checked override for overload as a whole.
         if e.func.info and not e.func.is_dynamic() and not e.is_overload:
-            found_base_method = self.check_method_override(e)
-            if e.func.is_explicit_override and found_base_method is False:
+            found_method_base_classes = self.check_method_override(e)
+            if (
+                e.func.is_explicit_override
+                and not found_method_base_classes
+                and found_method_base_classes is not None
+            ):
                 self.msg.no_overridable_method(e.func.name, e.func)
+            elif (
+                found_method_base_classes
+                and self.options.strict_override_decorator
+                and not e.func.is_explicit_override
```

Comment:
Similar to above -- what about `__init__` and `__new__`? I wonder if the logic could be shared in a helper, at least partially, since we have similar code in three places.